GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1778,7 +1765,16 @@ private void Unstage(bool canUseUnstageAll = true)
                         }
 
                         item.Staged = StagedStatus.WorkTree;
-                        unstagedFiles.Add(item);
+                        int index = unstagedFiles.FindIndex(i => i.Name == item.Name);
+
+                        if (index >= 0)
+                        {
+                            unstagedFiles[index] = item;
```

Comment:
This is not correct for the (constructed) opposite use case:
- change a file
- stage it
- delete it
- unstage

Reply:
will need a change

Reply:
That case was not handled correctly before either. With the modification, the status is seemingly correct when unstaging, but the worktree status is still deleted.
The problem is in GitModule.cs BatchUnstageFiles()
Ignoring for the moment

Reply:
I agree, the remaining is a very minor issue, which is not worth fixing IMO because the diff is displayed correctly.